zephyr/esp32s2/CMakeLists.txt

```diff
@@ -4,6 +4,9 @@ if(CONFIG_SOC_SERIES_ESP32S2)
 
   zephyr_compile_options(-fstrict-volatile-bitfields)
   zephyr_compile_definitions_ifndef(CONFIG_MCUBOOT CONFIG_APP_BUILD_USE_FLASH_SECTIONS)
+  zephyr_compile_definitions_ifdef(CONFIG_MCUBOOT BOOTLOADER_BUILD) # See Zephyr-ESP / ZEP-820
```

Comment:
What is the reason to include this (`NDEBUG`) only for ESP32-S2?
Would you also modifying all CMakeLists.txt to make these `BOOTLOADER_BUILD` definition in this same area? That would make it kind of common stuff among socs.

Reply:
yeah, this flag has been there since 4.1 hiding in the `sdkconfig.h`, and we keep copying it. I removed it and tested it with no issues now.